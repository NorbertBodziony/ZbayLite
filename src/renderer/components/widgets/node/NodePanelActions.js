import React from 'react'
import PropTypes from 'prop-types'

import Grid from '@material-ui/core/Grid'
import IconButton from '@material-ui/core/IconButton'

import RefreshIcon from '@material-ui/icons/Refresh'
import PowerSettingsIcon from '@material-ui/icons/PowerSettingsNew'

export const NodePanelActions = ({ onRestart, onPower }) => (
  <Grid container justify='flex-end'>
    <IconButton onClick={onRestart}>
      <RefreshIcon />
    </IconButton>
    <IconButton onClick={onPower}>
      <PowerSettingsIcon />
    </IconButton>
  </Grid>
)

NodePanelActions.propTypes = {
  onRestart: PropTypes.func.isRequired,
  onPower: PropTypes.func.isRequired
}

export default React.memo(NodePanelActions)
