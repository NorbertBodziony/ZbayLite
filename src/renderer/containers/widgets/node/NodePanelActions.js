import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelActions from '../../../components/widgets/node/NodePanelActions'

import nodeHandlers from '../../../store/handlers/node'

export const mapDispatchToProps = dispatch => bindActionCreators({
  onRestart: nodeHandlers.epics.restart,
  onPower: nodeHandlers.epics.togglePower
}, dispatch)

export default connect(null, mapDispatchToProps)(NodePanelActions)
