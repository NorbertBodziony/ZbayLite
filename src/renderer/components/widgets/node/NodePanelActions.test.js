import React from 'react'
import { shallow } from 'enzyme'

import { NodePanelActions } from './NodePanelActions'

describe('NodePanelActions', () => {
  it('renders component', () => {
    const result = shallow(
      <NodePanelActions onRestart={jest.fn()} onPower={jest.fn()} />
    )
    expect(result).toMatchSnapshot()
  })
})
