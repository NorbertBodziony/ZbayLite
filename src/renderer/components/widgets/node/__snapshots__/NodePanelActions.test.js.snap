// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodePanelActions renders component 1`] = `
<WithStyles(Grid)
  container={true}
  justify="flex-end"
>
  <WithStyles(IconButton)
    onClick={[MockFunction]}
  >
    <pure(RefreshIcon) />
  </WithStyles(IconButton)>
  <WithStyles(IconButton)
    onClick={[MockFunction]}
  >
    <pure(PowerSettingsNewIcon) />
  </WithStyles(IconButton)>
</WithStyles(Grid)>
`;
